server/channels/api4/user.go

```diff
@@ -3094,13 +3094,85 @@ func getChannelMembersForUser(c *Context, w http.ResponseWriter, r *http.Request
 		return
 	}
 
-	members, err := c.App.GetChannelMembersWithTeamDataForUserWithPagination(c.AppContext, c.Params.UserId, c.Params.Page, c.Params.PerPage)
-	if err != nil {
-		c.Err = err
+	// For backward compatibility purposes
+	if c.Params.Page != -1 {
+		cursor := &model.ChannelMemberCursor{
+			Page:    c.Params.Page,
+			PerPage: c.Params.PerPage,
+		}
+		members, err := c.App.GetChannelMembersWithTeamDataForUserWithPagination(c.AppContext, c.Params.UserId, cursor)
+		if err != nil {
+			c.Err = err
+			return
+		}
+
+		if err := json.NewEncoder(w).Encode(members); err != nil {
+			c.Logger.Warn("Error while writing response", mlog.Err(err))
+		}
 		return
 	}
 
-	if err := json.NewEncoder(w).Encode(members); err != nil {
+	// The new model streams data all in one shot.
+	pageSize := 100
+	fromChannelID := ""
+
+	// We have to write `[` and `]` separately because we want to stream the response.
+	// The internal API is paginated, but the client always needs to get the full data.
+	// Therefore, to avoid forcing the client to go through all the pages,
+	// we stream the full data from server side itself.
+	//
+	// Note that this means if an error occurs in mid-stream, the response won't be
+	// fully JSON.
+	if _, err := w.Write([]byte(`[`)); err != nil {
+		c.Logger.Warn("Error while writing response", mlog.Err(err))
+	}
```

Comment:
Here's a crazy idea -- set the content-type to `application/x-ndjson` and stream the response for each member (like we are), but then we don't have to manually write the `[],` array syntax. It would be able to handle errors, too. The client-side would need to be rewritten, but not by much.
The only downside (I think) would be it would cause trouble for other client implementations, but do we have any?

Reply:
Oooh brilliant idea! Given that this mode gets activated only on passing -1, I believe it's not a breaking change. We should also do this for the `getChannelsForUser` for v11.

Reply:
I like the idea, but you should check if mobile can support that content type before we use this. The web app's `Client4` will need changes for that as well, but mobile would likely have more problems with it due to backwards compatibility

Reply:
This is a new feature. Mobile doesn't even use this query param.

Reply:
I mean that we should make sure it's even possible for whatever library they use to handle newline-delimited JSON since this isn't as widely supported of a format, especially if we're planning on using it for existing APIs in a future release